sunpy/net/hek/hek.py

```diff
@@ -88,18 +95,78 @@ def search(self, *query):
             ndata.append(new)
 
         if len(ndata) == 1:
-            return self._download(ndata[0])
+            return HEKResponse(self._download(ndata[0]), client=self)
         else:
-            return self._merge(self._download(data) for data in ndata)
+            return HEKResponse(self._merge(self._download(data) for data in ndata), client=self)
 
     def _merge(self, responses):
         """ Merge responses, removing duplicates. """
         return list(unique(chain.from_iterable(responses), _freeze))
 
+    def fetch(self, *args, **kwargs):
+        return NotImplemented
+
+    @classmethod
+    def _attrs_module(cls):
+        return 'hek', 'sunpy.net.hek.attrs'
+
+    @classmethod
+    def _can_handle_query(cls, *query):
+        required = {attrs.Time}
+        optional = {i[1] for i in inspect.getmembers(attrs, inspect.isclass)} - required
+        qr = tuple()
+        for x in query:
+            if not isinstance(x, attrs.EventType):
+                qr += (x,)
+        return cls.check_attr_types_in_query(qr, required, optional)
+
+
+class HEKResponse(BaseQueryResponse):
+    """
+    A container for data returned from HEK searches.
+    """
+    def __init__(self, table=None, client=None):
+        """
+        table : `astropy.table.Table`
+        """
+        super().__init__()
+        self.table = table or astropy.table.Table()
+        self._client = client
+
+    @property
+    def client(self):
```

Comment:
Are these not defined on `BaseQueryResponse`?

Reply:
It is defined as an `abstractmethod`, so need to be defined here.

Reply:
But do we not have a class where we have created this class 3 times (JSOC, the two metadata clients) and they have the same code?

Reply:
Not only in jsoc, but we define it in all clients, like `vso`, `Generic` too. They all do the same, i.e., return `self._client`. We can't possibly do it in `BaseQueryResponse` becuase we don't initialize it with a `_client`.

Reply:
Now there's `BaseQueryResponeTable` class, not need to be done in every class.

Reply:
Thats good.